wp-graphql/wp-graphql
# test: lint and fix wpunit tests

What does this implement/fix? Explain your changes.
---------------------------------------------------
This PR lints and applies some basic fixes/refactors to existing WPUnit tests.
Work on #2408 exposed many cases of data being leaked between tests, however fixing the issues there makes PR review nearly unusable due to both the scope, and unrelated diffs from linting. It also makes merging/rebasing a PITA.

**There are also numerous cases where tests are _passing_ even though they are written incorrectly or there is a bug in the plugin, due to leaky data.**

#2408 is not the first PR where I've encountered this, and I believe this will make future work significantly more efficient. While this may cause some additional merge conflicts on already-open PRs, many of those already have merge conflicts that must be dealt with anyway.

### To test:
Add the following method to the bottom of `WPHelperTest.php` (so it runs last), and run `vendor/bin/codecept run wpunit -vvv`.
```php
public function testEnsureTypeCleanup() {
  $post_types = get_post_types();
  codecept_debug( $post_types );
  foreach ( $post_types as $post ) {
    codecept_debug( $post );
    codecept_debug( get_posts( [ 'post_type' => $post ] ) );
  }

  $taxonomies = get_taxonomies();
  codecept_debug( $taxonomies );
  foreach ( $taxonomies as $taxonomy ) {
    codecept_debug( $taxonomy );
    codecept_debug( get_terms( [ 'taxonomy' => $taxonomy ] ) );
  }

  codecept_debug( get_users() );
}
```
Only default WP post types and taxonomies should appear, the admin user, and _no_ posts or terms.

Does this close any currently open issues?
------------------------------------------
…


Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)


Any other comments?
-------------------
- While this touches many `wpunit` tests, it does not touch all of them, only those affected by bugs/leaky types discovered in #2408. This is intentional, in an attempt to limit as many merge conflicts as possible.


Where has this been tested?
---------------------------
**Operating System:** Ubuntu 20.04 (wsl2 + devilbox + PHP 8.0.19)

**WordPress Version:** 6.0.1
